### DISCUSSION TITLE
Lesson 18: Doubt regarding read storage in a view function

### DISCUSSION THREAD

**Discussion by ayushgupta0610** [Cyfrin/foundry-full-course-cu#1615] [General]
If the storage read costs gas, 100 units to be precise for a slot, how come the view functions, which also read from the storage, do not cost any gas?

**Comment by EngrPips** [comment]
You asked an interesting question, so reading from storage costs gas only while running a transaction. However, whenever you query the blockchain only to read a state on the blockchain, you would not be paying any gas to read from storage. I hope my explanation is good enough to make you understand; if anything still seems confusing, please let me know so I can explain it better

**Reply by ayushgupta0610** [reply]
I made an assumption that that might be the case but curious to know why is that the case. If you have any link that you'd like me to refer to, that would be helpful as well :)

**Reply by EngrPips** [reply]
What I know is it is so because transactions change the global state of the blockchain while querying the blockchain for already existing data doesn't change any state; hence, your transaction is not broadcasted, and you are getting the information from the node you are connected to but there might a whole paper explaining it somewhere, but right now I do not have a link to one
